server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java

```diff
@@ -181,25 +177,9 @@ private void testCase(Query query, CheckedConsumer<RandomIndexWriter, IOExceptio
         testCase(aggregationBuilder, query, buildIndex, verify, fieldType);
     }
 
-    private void testCase(CardinalityAggregationBuilder aggregationBuilder, Query query,
+    protected void testCase(CardinalityAggregationBuilder aggregationBuilder, Query query,
                           CheckedConsumer<RandomIndexWriter, IOException> buildIndex, Consumer<InternalCardinality> verify,
                           MappedFieldType fieldType) throws IOException {
-        Directory directory = newDirectory();
-        RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
-        buildIndex.accept(indexWriter);
-        indexWriter.close();
-
-        IndexReader indexReader = DirectoryReader.open(directory);
-        IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
-
-        CardinalityAggregator aggregator = createAggregator(aggregationBuilder, indexSearcher,
-            fieldType);
-        aggregator.preCollection();
-        indexSearcher.search(query, aggregator);
-        aggregator.postCollection();
-        verify.accept((InternalCardinality) aggregator.buildAggregation(0L));
-
-        indexReader.close();
-        directory.close();
+        super.testCase(aggregationBuilder, query, buildIndex, verify, fieldType);
```

Comment:
Creating a method with specific (and not generic) arguments is a shortcut that saves us from writing
```
   this.<CardinalityAggregationBuilder, InternalCardinality>testCase(.....)
```

at every test case. This method simply passes the arguments to the generic `AggregatorTestCase.testCase()` method. I don't have a strong opinion about which option is best and I am happy go with any of the two methods.